src/QueryDataTable.php

```diff
@@ -156,10 +164,21 @@ public function prepareCountQuery(): QueryBuilder
         $builder = clone $this->query;
 
         if ($this->isComplexQuery($builder)) {
+            $builder->select(DB::raw('1'));
+
+            if ($this->ignoreSelectInCountQuery || ! $this->isComplexQuery($builder)) {
```

Comment:
Maybe we remove and fix the negated boolean?

```suggestion
            if ($this->ignoreSelectInCountQuery) {
```

Reply:
Oh, test failed when it was removed... checking further.

Reply:
Test fixed by setting false in has no select.

Reply:
# It seems that PHPStan does not understand that this change the `$builder` instance : 
```php
$builder->select(DB::raw('1'));
```

We are first checking if the query with the select is a complexe query,

if true then

we are overriding the selects, to only select '1', removing everything.

then, rechecking if the query is still complexe (if anything outside the selects use a order, union, group by etc...)

If the query is no more complexe, we use the new query without selects, else, we use it with the selects as the order, group by etc... might use them